src/metrics/timingMetric.js

```diff
@@ -0,0 +1,43 @@
+import {VALID_TYPES} from './metrics';
+
+function typeOrError(type) {
+  if (VALID_TYPES.includes(type)) {
+    return type;
+  }
+  throw new Error('invalid type for a metric supplied');
+}
+
+export default function newMetric(metricType, {fields, tags} = {
+  fields: [],
+  tags: []
+}, stop = undefined, start = undefined) {
+  const type = typeOrError(metricType);
+  const startTime = start || new Date();
+
+  return {
+    addField(key, value) {
+      fields[key] = value;
+      return this;
+    },
+    addTag(key, value) {
+      tags[key] = value;
+      return this;
+    },
+    get() {
+      const duration = (stop || new Date()) - startTime;
+      return {
+        type,
+        duration,
+        tags,
+        fields
```

Comment:
Same remark here regarding passing by reference. Might be an idea to have copy these on return. Imagine for example these being placed on a queue for batching to Influx. That could introduce some interesting side-effects based on what you change in the fields/tags.